test/gctime_test.rb

```diff
@@ -3,22 +3,28 @@
 require "test_helper"
 
 class GCTimeTest < Minitest::Test
-  def setup
-    GCTime.enable
+  def test_total_time_is_increasing
+    initial_time = GC.total_time
+    100.times { "a" * 100 }
+    4.times { GC.start }
+    assert_operator initial_time, :<, GC.total_time
   end
 
-  def teardown
-    GCTime.disable
+  def test_total_time_is_an_integer
+    assert_instance_of Integer, GC.total_time
   end
 
-  def test_total_time_is_increasing
-    initial_time = GCTime.total_time
-    100.times { "a" * 100 }
-    4.times { GC.start }
-    refute_equal initial_time, GCTime.total_time
+  def test_gc_stat_time_is_a_numeric
+    # JRuby and TruffleRuby expose it as a Float number of milliseconds
+    # MRI 3.1 as an integer number of milliseconds
+    assert_kind_of Numeric, GC.stat(:time)
+    assert_kind_of Numeric, GC.stat[:time]
   end
 
-  def test_total_time_is_a_float
-    assert_instance_of Float, GCTime.total_time
+  def test_measure_total_time
+    GC.measure_total_time = false
+    GC.measure_total_time
```

Comment:
This doesn't seem to test anything. Did you mean:

```suggestion
    assert(GC.measure_total_time)
```

Reply:
It kinda asserts that it doesn't reaise, but yeah I could make it more explictly.